src/main/java/org/wso2/carbon/connector/utils/EmailConstants.java

```diff
@@ -0,0 +1,95 @@
+    public static final String CONNECTION_TYPE = "connectionType";
+    public static final String HOST = "host";
+    public static final String PORT = "port";
+    public static final String USERNAME = "username";
+    public static final String PASSWORD = "password";
+    public static final String FOLDER = "folder";
+    public static final String EMAIL_ID = "emailID";
+    public static final String EMAIL_INDEX = "emailIndex";
+    public static final String ATTACHMENT_INDEX = "attachmentIndex";
+    public static final String CONNECTION = "connection";
+    public static final String DELETE_AFTER_RETRIEVE = "deleteAfterRetrieve";
+    public static final String RECEIVED_SINCE = "receivedSince";
+    public static final String RECEIVED_UNTIL = "receivedUntil";
+    public static final String SENT_SINCE = "sentSince";
+    public static final String SENT_UNTIL = "sentUntil";
+    public static final String SUBJECT_REGEX = "subjectRegex";
+    public static final String FROM_REGEX = "fromRegex";
+    public static final String OFFSET = "offset";
+    public static final String LIMIT = "limit";
+    public static final String READ_TIMEOUT = "readTimeout";
+    public static final String WRITE_TIMEOUT = "writeTimeout";
+    public static final String CONNECTION_TIMEOUT = "connectionTimeout";
+    public static final String REQUIRE_TLS = "requireTLS";
+    public static final String CHECK_SERVER_IDENTITY = "checkServerIdentity";
+    public static final String TRUSTED_HOSTS = "trustedHosts";
+    public static final String SSL_PROTOCOLS = "sslProtocols";
+    public static final String CIPHER_SUITES = "cipherSuites";
+    public static final String MAX_ACTIVE_CONNECTIONS = "maxActiveConnections";
+    public static final String MAX_IDLE_CONNECTIONS = "maxIdleConnections";
+    public static final String MAX_WAIT_TIME = "maxWaitTime";
+    public static final String MAX_EVICTION_TIME = "minEvictionTime";
+    public static final String EVICTION_CHECK_INTERVAL = "evictionCheckInterval";
+    public static final String EXHAUSTED_ACTION = "exhaustedAction";
+    public static final String INITIALISATION_POLICY = "initialisationPolicy";
+    public static final String DISABLE_POOLING = "disablePooling";
+    //Java Mail API Constants
+    public static final String CONTENT_TRANSFER_ENCODING_HEADER = "Content-Transfer-Encoding";
+    // Default configuration values
+    public static final String DEFAULT_CONTENT_TYPE = "text/html";
+    public static final String DEFAULT_ENCODING = "UTF-8";
+    public static final String DEFAULT_CONTENT_TRANSFER_ENCODING = "Base64";
+    public static final String DEFAULT_SOCKETFACTORY_CLASS = "javax.net.ssl.SSLSocketFactory";
+    public static final String DEFAULT_SOCKETFACTORY_FALLBACK = "false";
+    public static final String DEFAULT_FOLDER = "INBOX";
+    public static final int DEFAULT_OFFSET = 0;
+    //TODO: Should this be infinite?
+    public static final int DEFAULT_LIMIT = 10;
```

Comment:
If limit is not specified, shall we set it to -1 and retrieve all the emails in the box?

Reply:
Fixed